client-react/src/features/todos/components/TodoActions/TodoActions.tsx

```diff
@@ -1,25 +1,34 @@
+import { Button } from "@mui/material"
+import { useAppDispatch } from "../../../../app/hooks"
+import { TodosCollectionActions } from "../../../../store/actions"
 import { Todo } from "../../models/todo.model"
-import { TodoAction } from "../../models/todoAction.model"
-
+import { TodoModalsActions } from "../../store/actions"
 
 interface TodoActionsProps {
   todo: Todo,
-  todoActions?: Array<TodoAction>
 }
 
-export const TodoActions = ({todo, todoActions}: TodoActionsProps) => {
+export const TodoActions = ({todo}: TodoActionsProps) => {
+  const dispatch = useAppDispatch()
+
+  const openEditModal = (todo: Todo) => {
+    dispatch(TodoModalsActions.openTodoModal(todo))
+  }
+
+  const toggleTodo = ({id, completed}: Todo) => {
+    dispatch(TodosCollectionActions.updateSingle({ id: id, todo: {completed: !completed} }))
+  }
+
+  const editTodo = true;
+
   return(
-    <div>
-      { 
-        (todoActions) &&
-        (todoActions.map(item => {
-          return(
-            <button onClick={() => item.action(todo)}>
-              {item.title}
-            </button>
-          )
-        }))
-      }
-    </div>
+    <>
+      {(editTodo) &&<Button onClick={() => openEditModal(todo)}>
+        Edit Todo
+      </Button>}
+      {(toggleTodo) && <Button onClick={() => toggleTodo(todo)}>
```

Comment:
Why do we render toggle action if there's a toggleTodo function? This always will be evaluated to true.

Reply:
ok zoomer